### PR TITLE
Add NetworkACL feature to AzureRM.Storage

### PR DESCRIPTION
**Pull request by blueww**
## Description
Add NetworkACL feature to AzureRM.Storage
The design doc can be found in [PowerShell NetworkACL Deign](https://microsoft.sharepoint.com/teams/XClientShanghai/_layouts/15/guestaccess.aspx?guestaccesstoken=M7LS4xrB7SUuIg3AZWD4zhpoX0t5TTxIl%2bl9nc2sRRc%3d&docid=2_1dd1334bf31fe4b9c8055f39875d65eb6&rev=1). Mark and I have reviewed it and get agreement on it.

The code change has passed storage regression.
And Unit test is also added

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [Y ] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [Y ] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [Y ] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [ Y] Title of the pull request is clear and informative.
- [ Y] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [Y ] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [ Y] Pull request includes test coverage for the included changes.
- [ Y] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [ Y] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [ Y] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [Y ] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [Y ] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [Y ] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by blueww** [comment]
On Demand Test: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1682/

File: src/ResourceManager/Storage/Commands.Management.Storage/StorageAccount/AddAzureStorageAccountNetworkACLRule.cs [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using Microsoft.Azure.Management.Storage;
+using Microsoft.Azure.Management.Storage.Models;
+using StorageModels = Microsoft.Azure.Management.Storage.Models;
+using Microsoft.Azure.Commands.Management.Storage.Models;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Commands.Management.Storage
+{
+    [Cmdlet(VerbsCommon.Add, StorageAccountACLRuleNounStr, SupportsShouldProcess = true)]
```
Comment (markcowl) [inline src/ResourceManager/Storage/Commands.Management.Storage/StorageAccount/AddAzureStorageAccountNetworkACLRule.cs]: need to set DefaultParameterSetName

File: src/ResourceManager/Storage/Commands.Management.Storage/StorageAccount/RemoveAzureStorageAccountNetworkACLRule.cs [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+using Microsoft.Azure.Management.Storage;
+using Microsoft.Azure.Management.Storage.Models;
+using StorageModels = Microsoft.Azure.Management.Storage.Models;
+using Microsoft.Azure.Commands.Management.Storage.Models;
+using System.Collections.Generic;
+using System;
+
+namespace Microsoft.Azure.Commands.Management.Storage
+{
+    [Cmdlet(VerbsCommon.Remove, StorageAccountACLRuleNounStr, SupportsShouldProcess = true)]
```
Comment (markcowl) [inline src/ResourceManager/Storage/Commands.Management.Storage/StorageAccount/RemoveAzureStorageAccountNetworkACLRule.cs]: Need to specify DefaultParameterSetName

**Comment by markcowl** [comment]
This looks good once these changes are made

File: src/ResourceManager/Storage/Commands.Management.Storage/StorageAccount/RemoveAzureStorageAccountNetworkACLRule.cs (hunk 2 above)
Comment (blueww) [inline src/ResourceManager/Storage/Commands.Management.Storage/StorageAccount/RemoveAzureStorageAccountNetworkACLRule.cs]: Default Parameter Set added

**Comment by blueww** [comment]
on Demand test update in : http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1697/

**Comment by blueww** [comment]
Change log added.
On-demand test updated in : http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1701/

**Pull request merged**